feature/experimental/gribi/ate_tests/route_addition_during_failover_test/route_addition_during_failover_test.go

```diff
@@ -0,0 +1,655 @@
+	a.PrefixLength = ygot.Uint8(uint8(ipv4PrefixLen))
+}
+
+// configureATE configures a single ATE layer 3 interface.
+func configureATE(t *testing.T, top *ondatra.ATETopology, atePort *ondatra.Port, Name string, vlanID uint16, dutIPv4, ateIPv4 string) {
+	t.Helper()
+
+	i := top.AddInterface(Name).WithPort(atePort)
+	if vlanID != 0 {
+		i.Ethernet().WithVLANID(vlanID)
+	}
+	i.IPv4().WithAddress(ateIPv4)
+	i.IPv4().WithDefaultGateway(dutIPv4)
+}
+
+// awaitTimeout calls a fluent client Await, adding a timeout to the context.
+func awaitTimeout(ctx context.Context, c *fluent.GRIBIClient, t testing.TB, timeout time.Duration) error {
+	subctx, cancel := context.WithTimeout(ctx, timeout)
+	defer cancel()
+	return c.Await(subctx, t)
+}
+
+// testArgs holds the objects needed by a test case.
+type testArgs struct {
+	ctx        context.Context
+	client     *fluent.GRIBIClient
+	dut        *ondatra.DUTDevice
+	ate        *ondatra.ATEDevice
+	top        *ondatra.ATETopology
+	electionID gribi.Uint128
+}
+
+// createTrafficFlow generates traffic flow from source network to destination network via
+// srcEndPoint to dstEndPoint and checks for packet loss.
+func createTrafficFlow(t *testing.T, ate *ondatra.ATEDevice, top *ondatra.ATETopology, flowArgs *flowArgs) *ondatra.Flow {
+	ethHeader := ondatra.NewEthernetHeader()
+	ipv4Header := ondatra.NewIPv4Header()
+	ipv4Header.DstAddressRange().
+		WithMin(flowArgs.flowStartAddress).
+		WithMax(flowArgs.flowEndAddress).
+		WithCount(flowArgs.flowCount)
+
+	srcEndPoint := top.Interfaces()["src"]
+	dstEndPoint := []ondatra.Endpoint{}
+	for intf, intfData := range top.Interfaces() {
+		if intf != "src" {
+			dstEndPoint = append(dstEndPoint, intfData)
+		}
+	}
+
+	flow := ate.Traffic().NewFlow("Flow").
+		WithSrcEndpoints(srcEndPoint).
+		WithDstEndpoints(dstEndPoint...).
+		WithHeaders(ethHeader, ipv4Header)
+	return flow
+}
+
+// findSecondaryController finds out primary and secondary controllers.
+func findSecondaryController(t *testing.T, dut *ondatra.DUTDevice, controllers []string) (string, string) {
+	t.Helper()
+
+	var primary, secondary string
+	for _, controller := range controllers {
+		role := gnmi.Get(t, dut, gnmi.OC().Component(controller).RedundantRole().State())
+		t.Logf("Component(controller).RedundantRole().Get(t): %v, Role: %v", controller, role)
+		switch role {
+		case secondaryController:
+			secondary = controller
+		case primaryController:
+			primary = controller
+		default:
+			t.Fatalf("Expected controller %s to be active or standby, got %v", controller, role)
+		}
+	}
+	if secondary == "" || primary == "" {
+		t.Fatalf("Expected non-empty primary and secondary Controller, got primary: %v, secondary: %v", primary, secondary)
+	}
+	t.Logf("Detected primary: %v, secondary: %v", primary, secondary)
+	return secondary, primary
+}
+
+// switchoverReady is to check if controller is ready for switchover.
+func switchoverReady(t *testing.T, dut *ondatra.DUTDevice, controller string) {
```

Comment:
better to start the function name with a verb. "awaitSwitchoverReady"?

Reply:
updated, thank you